library/src/blas2/rocblas_hpr.hpp

```diff
@@ -0,0 +1,125 @@
+/* ************************************************************************
+ * Copyright 2016-2020 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#pragma once
+#include "handle.h"
+#include "rocblas.h"
+
+template <typename T, typename U>
+__device__ void
+    hpr_kernel_calc(bool upper, rocblas_int n, U alpha, const T* x, rocblas_int incx, T* AP)
+{
+    rocblas_int tx = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    rocblas_int ty = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+    int index = upper ? ((ty * (ty + 1)) / 2) + tx : ((ty * (2 * n - ty + 1)) / 2) + (tx - ty);
+
+    if(upper ? ty < n && tx < ty : tx < n && ty < tx)
+        AP[index] += alpha * x[tx * incx] * conj(x[ty * incx]);
+    else if(tx == ty && tx < n)
+    {
+        T x_real  = std::real(x[tx * incx]);
+        T x_imag  = std::imag(x[tx * incx]);
```

Comment:
T is complex and we fetch the real part as a complex?

Reply:
Good catch,  will fix.